### ISSUE TITLE
test_prevote_reboot_minority_followers failed

### ISSUE THREAD

**Issue by zhangjinpeng1987** [tikv/tikv#4029]
## Bug Report

**What version of Rust are you using?**
rustc 1.29.0-nightly (4f3c7a472 2018-07-17)

**What operating system and CPU are you using?**
MacOS

**What did you do?**
run `make dev`

**What did you expect to see?**
test failed

**What did you see instead?**
```
thread 'raftstore::test_prevote::test_prevote_reboot_minority_followers' panicked at 'assertion failed: `(left == right)`
  left: `true`,
 right: `false`: Sends a PreVote or PreVoteResponse during failure.', tests/integrations/raftstore/test_prevote.rs:90:9
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::print
             at libstd/sys_common/backtrace.rs:71
             at libstd/sys_common/backtrace.rs:59
   2: std::panicking::default_hook::{{closure}}
             at libstd/panicking.rs:211
   3: std::panicking::default_hook
             at libstd/panicking.rs:227
   4: <std::panicking::begin_panic::PanicPayload<A> as core::panic::BoxMeUp>::get
             at libstd/panicking.rs:475
   5: std::panicking::continue_panic_fmt
             at libstd/panicking.rs:390
   6: std::panicking::try::do_call
             at libstd/panicking.rs:345
   7: integrations::raftstore::test_prevote::test_prevote
             at tests/integrations/raftstore/test_prevote.rs:90
   8: integrations::raftstore::test_prevote::test_prevote_reboot_minority_followers
             at tests/integrations/raftstore/test_prevote.rs:212
   9: integrations::__test::TESTS::{{closure}}
             at tests/integrations/raftstore/test_prevote.rs:209
  10: core::ops::function::FnOnce::call_once
             at /Users/travis/build/rust-lang/rust/src/libcore/ops/function.rs:223
  11: <F as alloc::boxed::FnBox<A>>::call_box
             at libtest/lib.rs:1454
             at /Users/travis/build/rust-lang/rust/src/libcore/ops/function.rs:223
             at /Users/travis/build/rust-lang/rust/src/liballoc/boxed.rs:640
  12: panic_unwind::dwarf::eh::read_encoded_pointer
             at libpanic_unwind/lib.rs:106
```

**Comment by Hoverbear** [comment]
We've seen this several times, it occurs because we cannot start watching soon enough to see the message sent. 😓

**Comment by hicqu** [comment]
Closed by #6320.
